### PR TITLE
TfLite Stablehlo add dot_general op basic version

### PR DESCRIPTION
**Pull request by nishantsarda-mcw**
- Adds Eigen Implementation of dot_general op, supporting 3d input (batch 2d matric)
- Adds unit test for F32, BF16 and F16.

### CODE REVIEW

**Comment by rascani** [comment]
When will this PR be updated with requested changes?

**Comment by nishantsarda-mcw** [comment]
Hi RJ, I have a doubt, so in the base branch of dot_general right now it doesn't have struct Opdata but later dot_general requires struct Opdata (where temporary tensors are needed). and because of this lot of changes are being seen when you review. 
But now if I do changes in the 1st branch(base_branch) i.e add the struct Opdata, and do a new commit, you will still see the diff between the earlier commit and this commit in the 1st branch. 
one more option is that I do git commit amend, here you will see proper code but will lose the history. and from 2nd branch(constraint_branch), I can do rebase then it will only show the latest changes and so on for 3rd and 4th branch. I am sorry, I know you have already reviewed the 1st branch and this git amend will remove the history, but there is this that the base branch only had 2 comments, is this fine?
Can you suggest which way should I proceed?

**Comment by rascani** [comment]
Please make a new commit, only addressing the 2 comments on this PR. I will review that rework and we can merge this change. Then we'll update #20 onto the tflite_shlo_dot_general_op branch and we can proceed with the review of #20.

File: tensorflow/lite/core/c/builtin_op_data.h [hunk 1]
```diff
@@ -592,6 +593,26 @@ typedef struct {
...
+  int num_rhs_batching_dimensions;
+  int64_t lhs_contracting_dimensions
+      [TFLITE_STABLEHLO_DOT_GENERAL_PARAMS_MAX_DIMENSION_COUNT];
+  int num_lhs_contracting_dimensions;
+  int64_t rhs_contracting_dimensions
+      [TFLITE_STABLEHLO_DOT_GENERAL_PARAMS_MAX_DIMENSION_COUNT];
+  int num_rhs_contracting_dimensions;
+  int num_precision_configs;
+  uint32_t
+      precision_config[TFLITE_STABLEHLO_DOT_GENERAL_PARAMS_MAX_DIMENSION_COUNT];
```
[the remark this answers is not in the document]
Comment (nishantsarda-mcw) [reply tensorflow/lite/core/c/builtin_op_data.h]: Addressed in a new commit [13ff077](https://github.com/MCW-Dev/tensorflow/commit/13ff0776ce9bd385280778d530e4c8e0c3d8b153)

File: tensorflow/lite/kernels/BUILD [hunk 2]
```diff
@@ -887,6 +888,7 @@ cc_library(
         ":variable_op_kernels",
         "@fft2d",
         "@ruy//ruy/profiler:instrumentation",
+        "//tensorflow/core/platform:bfloat16",
```
[the remark this answers is not in the document]
Comment (nishantsarda-mcw) [reply tensorflow/lite/kernels/BUILD]: Addressed in a new commit [13ff077](https://github.com/MCW-Dev/tensorflow/commit/13ff0776ce9bd385280778d530e4c8e0c3d8b153)

**Review by rascani — APPROVED** [review]

**Pull request merged**